### DISCUSSION TITLE
UniteNodes

### DISCUSSION THREAD

**Discussion by bhnizar** [Dlubal-Software/RFEM_Python_Client#368] [Q&A] [answered]
Hello,

I want to merge duplicate nodes. For this, I used the following script. However, I get an error message that blocks the command and I can't understand where the problem comes from. 

Could you please help me?

![Capture](https://github.com/Dlubal-Software/RFEM_Python_Client/assets/108261434/27bcb537-1c5f-4097-803a-e101ec7030f1)

**Comment by OndraMichal** [comment]
Hi @bhnizar,
the problem is the finish_modification function needs to be placed after all changes in the model i.e. before calculate_all function or at the very end of the script.

**Reply by bhnizar** [reply]
Hi @OndraMichal 
Thank you for your answer.
I placed the finish_modification after the ModelCheck.UniteNodes, but I'm still getting the same error message (see screenshot). 
Could you please provide an explanation?
![Capture](https://github.com/Dlubal-Software/RFEM_Python_Client/assets/108261434/8fa901e1-1918-4a5f-bce1-66546f7a8a73)

**Comment by bhnizar** [comment]
Hi @OndraMichal  @heetrojivadiya , 
To help you identify the problem I attach model py and the rf6 model.
Thanks in advance for your help :)

**Reply by heetrojivadiya** [reply]
Hello @bhnizar, 

Thank you for sharing code. I have checked in WS core and unfortunately this `model_check__get_object_groups_operation()` method is currently not available. 

However, you can omit duplicate nodes error during calculation by assigning correct numbers during member creation.

**Reply by bhnizar** [reply]
Okay , are you scheduled to implement this feature soon?

**Reply by heetrojivadiya** [reply, marked as answer]
Yes, the ToDo#8140 is created for it and currently it is in review. When it will be ready then we will let you know. 

If you just want to find identical nodes then you can use below script for your model.
```py
from RFEM.Tools.GetObjectNumbersByType import GetAllObjects
from RFEM.initModel import Model

Model(False, 'ombrière')
objects, imports = GetAllObjects()

nodes = []
for item in objects:
    dict = {}
    if 'Node(' in item:
        item = eval(item.replace('Node(params=', '').replace(')\n', ''))
        dict['no'] = item.get('no')
        dict['x'] = round(item.get('global_coordinate_1'), 3)
        dict['y'] = round(item.get('global_coordinate_2'), 3)
        dict['z'] = round(item.get('global_coordinate_3'), 3)
        nodes.append(dict)

identical_nodes = []
for i in nodes:
    for j in nodes[int(i['no']):]:
        if i != j and i['x'] == j['x'] and i['y'] == j['y'] and i['z'] == j['z']:
            identical_nodes.append((i['no'],j['no']))

print(identical_nodes)
```
Output:
```py
[(4, 47), (11, 52), (18, 57), (25, 62), (32, 67), (39, 72)]
```

Meanwhile if you need any other assistance regarding this issue or other WS problems then feel free to contact us.
